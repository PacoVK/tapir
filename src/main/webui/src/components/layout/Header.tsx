import React from "react";
import AppBar from "@mui/material/AppBar";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";
import { useUserContext } from "../context/UserContext";
import {
  Avatar,
  Unstable_Grid2 as Grid,
  Button,
  Menu,
  MenuItem,
} from "@mui/material";
import { deepOrange, orange } from "@mui/material/colors";
import LaunchIcon from "@mui/icons-material/Launch";

const Header = () => {
  const { user } = useUserContext();
  const [menuAchor, setMenuAchor] = React.useState<null | HTMLElement>(null);
  const open = Boolean(menuAchor);
  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setMenuAchor(event.currentTarget);
  };
  const handleClose = () => {
    setMenuAchor(null);
  };

  const openTapirInNewTab = () => {
    handleClose();
    window.open(
      process.env.REACT_APP_TAPIR_DOCS,
      "_blank",
      "noopener,noreferrer",
    );
  };

  const logout = async () => {
    handleClose();
    await fetch("logout").finally(() => window.location.replace("/"));
  };
  return (
    <AppBar position="static" sx={{ backgroundColor: "#474747" }}>
      <Grid container spacing={3} sx={{ flexGrow: 1 }}>
        <Grid xs={4} mdOffset="auto"></Grid>
        <Grid xs={4} xsOffset={0} mdOffset={0} alignSelf={"center"}>
          <Toolbar disableGutters>
            <Typography
              variant="h6"
              noWrap
              component="a"
              href="/"
              sx={{
                display: "flex",
                fontFamily: "monospace",
                fontWeight: 700,
                letterSpacing: ".3rem",
                color: "inherit",
                textDecoration: "none",
                margin: "auto",
                overflow: "visible",
              }}
            >
              Tapir - Private Terraform Registry
            </Typography>
          </Toolbar>
        </Grid>
        <Grid xs={4} alignSelf={"center"}>
          <Avatar
            onClick={handleClick}
            component={Button}
            id="menu-button"
            aria-controls={open ? "menu-menu" : undefined}
            aria-haspopup="true"
            aria-expanded={open ? "true" : undefined}
            sx={{
              ":hover": {
                cursor: "pointer",
                bgcolor: deepOrange[500],
              },
              bgcolor: orange[500],
              float: "right",
              marginRight: "2vw",
              minWidth: "unset",
            }}
          >
            {user.name ? user.name.charAt(0).toUpperCase() : undefined}
          </Avatar>
          <Menu
            id="user-menu"
            anchorEl={menuAchor}
            open={open}
            onClose={handleClose}
            MenuListProps={{
              "aria-labelledby": "menu-button",
            }}
          >
            <MenuItem onClick={openTapirInNewTab}>
              About
              <LaunchIcon sx={{ ml: "5px" }} />
            </MenuItem>
            <MenuItem onClick={logout}>Logout</MenuItem>
          </Menu>
        </Grid>
      </Grid>
    </AppBar>
  );
};

export default Header;

/*

 */
