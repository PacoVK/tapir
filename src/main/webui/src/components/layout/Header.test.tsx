import { render } from "@testing-library/react";
import Header from "./Header";

describe("<Header /> spec", () => {
  it("renders the Header", () => {
    const view = render(<Header />);
    expect(view).toMatchSnapshot();
  });
});
