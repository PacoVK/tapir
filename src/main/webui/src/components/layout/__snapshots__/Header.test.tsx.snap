// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> spec renders the Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1i39xt3-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-3 css-3ijkck-MuiGrid2-root"
        >
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1r8wmwf-MuiGrid2-root"
          />
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-109lejs-MuiGrid2-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
            >
              <a
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1hurzvz-MuiTypography-root"
                href="/"
              >
                Tapir - Private Terraform Registry
              </a>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-ktgzga-MuiGrid2-root"
          >
            <button
              aria-haspopup="true"
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-cadykg-MuiButtonBase-root-MuiButton-root-MuiAvatar-root"
              id="menu-button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1i39xt3-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-3 css-3ijkck-MuiGrid2-root"
      >
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1r8wmwf-MuiGrid2-root"
        />
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-109lejs-MuiGrid2-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
          >
            <a
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1hurzvz-MuiTypography-root"
              href="/"
            >
              Tapir - Private Terraform Registry
            </a>
          </div>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-ktgzga-MuiGrid2-root"
        >
          <button
            aria-haspopup="true"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-cadykg-MuiButtonBase-root-MuiButton-root-MuiAvatar-root"
            id="menu-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
